Tests/DiffableDataSourceTests/DiffableTests.swift

```diff
@@ -57,24 +57,28 @@ final class DiffableTests: XCTestCase {
     func testReplaceWithNonUniqueSectionIdentifiers() {
         let content: [DiffableSection<Section, Row>] = [topSection, topSection]
 
-        XCTAssertThrowsError(try dataSource.replace(with: content)) {
-            XCTAssertEqual($0 as? DiffableError, .sectionsNotUnique)
+        Dependency.logFault = { error in
+            XCTAssertEqual(error, .sectionsNotUnique)
         }
+
+        dataSource.replace(with: content)
```

Comment:
You could fix the problem of the test not failing when there is no error like this:
```suggestion
        dataSource.replace(with: content)

        XCTAssertEqual(loggedFault, .sectionsNotUnique)
```

Assuming that you add this field to `DiffableTests`:
```swift
    private var loggedFault: DiffableError?
```

and you add this to `func setUpWithError()`
```swift
        loggedFault = nil
        Dependency.logFault = { [weak self] error in
            self?.loggedFault = error
        }
```